File: typescript/src/filter.ts
import {listService} from "./listService";
import {isFailure, isSuccess, Result} from "for-comprehension-ts";

const results: Result<number[], string>[] = [2, -1, 4].map(listService)

const rights: number[][] = results.filter(isSuccess)
                                  .map(success => success.value)

console.log(rights)

const lefts: string[] = results.filter(isFailure)
                               .map(failure => failure.error)

console.log(lefts)

File: swift/Filter.swift

func filter() {
    printName()

    let results = [2, -1, 4].map(listService)

    let succeeds: [[Int]] = results.compactMap { try? $0.get() }

    print(succeeds)

    let errors: [String] = results.compactMap { result in
        if case .failure(let error) = result { return error } else { return nil }
    }
    print(errors)
}
